javascriptv3/example_code/bedrock-runtime/scenarios/converse_tool_scenario/converse-tool-scenario.js

```diff
@@ -0,0 +1,314 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// snippet-start:[Bedrock.ConverseTool.javascript.Scenario]
+
+/* Before running this JavaScript code example, set up your development environment, including your credentials.
+This demo illustrates a tool use scenario using Amazon Bedrock's Converse API and a weather tool.
+The script interacts with a foundation model on Amazon Bedrock to provide weather information based on user
+input. It uses the Open-Meteo API (https://open-meteo.com) to retrieve current weather data for a given location.*/
+
+import {
+  Scenario,
+  ScenarioAction,
+  ScenarioInput,
+  ScenarioOutput,
+  //} from "@aws-doc-sdk-examples/lib/scenario/index.js";
+} from "../../../libs/scenario/index.js";
+import {
+  BedrockRuntimeClient,
+  ConverseCommand,
+} from "@aws-sdk/client-bedrock-runtime";
+
+import { parseArgs } from "node:util";
+import { fileURLToPath } from "node:url";
+import { dirname } from "node:path";
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+import data from "./questions.json" with { type: "json" };
+import toolConfig from "./tool_config.json" with { type: "json" };
+
+const systemPrompt = [
+  {
+    text:
+      "You are a weather assistant that provides current weather data for user-specified locations using only\n" +
+      "the Weather_Tool, which expects latitude and longitude. Infer the coordinates from the location yourself.\n" +
+      "If the user provides coordinates, infer the approximate location and refer to it in your response.\n" +
+      "To use the tool, you strictly apply the provided tool specification.\n" +
+      "\n" +
+      "- Explain your step-by-step process, and give brief updates before each step.\n" +
+      "- Only use the Weather_Tool for data. Never guess or make up information. \n" +
+      "- Repeat the tool use for subsequent requests if necessary.\n" +
+      "- If the tool errors, apologize, explain weather is unavailable, and suggest other options.\n" +
+      "- Report temperatures in °C (°F) and wind in km/h (mph). Keep weather reports concise. Sparingly use\n" +
+      "  emojis where appropriate.\n" +
+      "- Only respond to weather queries. Remind off-topic users of your purpose. \n" +
+      "- Never claim to search online, access external data, or use tools besides Weather_Tool.\n" +
+      "- Complete the entire process until you have all required data before sending the complete response.",
+  },
+];
+const tools_config = toolConfig;
+
+/// Starts the conversation with the user and handles the interaction with Bedrock.
+async function askQuestion(userMessage) {
+  // The maximum number of recursive calls allowed in the tool use function.
+  // This helps prevent infinite loops and potential performance issues.
+  const max_recursions = 5;
```

Comment:
If we're going to use Washington State and France, we may need a higher recursion number than 5.